### ISSUE TITLE
[css-anchor-position-1] Automatic alignment / justification

### ISSUE THREAD

**Issue by xiaochengh** [w3c/csswg-drafts#9269]
In the [Anchor Positioning Exploration](https://fantasai.inkedblade.net/style/specs/css-anchor-exploration/) proposal, the used position area implies the [default alignment / justification](https://fantasai.inkedblade.net/style/specs/css-anchor-exploration/#area-alignment). For example, if the center track is used, then the default alignment is `anchor-center`.

Now that we agreed to add `inset-area` to the spec (https://github.com/w3c/csswg-drafts/issues/9145#issuecomment-1690207503), it also seems useful to further adopt default alignment to the current spec.

## Proposal

When the value of `align-self` / `justify-self` is `normal`, and the `inset-area` on the corresponding axis contains the `center` track, then `normal` behaves as `anchor-center`.

## Note

The [original proposal](https://fantasai.inkedblade.net/style/specs/css-anchor-exploration/#area-alignment) also sets default alignment for the `start` and `end` tracks, but I don't think they are still needed in the context of `inset-area`, because e.g. `inset-area: start` will be resolved into `inset-something: anchor(start)`, which already automatically align the element with the start edge of the anchor; same for the `end` track.

**Comment by kizu** [comment]
One thing I want to note regarding the automatic alignment: it would be very nice to support the transitions for when the element switches tracks.

I'm currently playing with the proposed `position-area`, and in my [latest experiment](https://codepen.io/kizu/pen/zYMmVJd) I implemented the alignment by nested elements and grid alignment. And while the inset properties have a proper transition, when the track change happens without a transition things do not look good:

https://github.com/w3c/csswg-drafts/assets/177485/aba2ca78-dd49-40f3-b7d2-09555d754c59

In the above video we can see how the alignment change without a transition when the element's position changes from `top / all` to `top / right`, as at first it wants to be at the center of the anchor horizontally, but then to be at the left.

Basically, we'd want the alignment to be a part of the final `inset-area` value or something, so if we'd have a transition for it, it would work well with the change in track alignment as well.

(I'll try to emulate this transition on my experiment, though not entirely sure if I would succeed; if so — I would update this comment).
